packages/stryker/src/MutatorOrchestrator.ts

```diff
@@ -130,6 +130,27 @@ export default class MutatorOrchestrator {
       }
     });
 
-    return mutants;
+    return this.reduceMutantSize(mutants, this.options.proportionOfMutantsToUse);
+  }
+
+  private reduceMutantSize(mutants: Mutant[], proportionToKeep: number) {
+    const mutantMap = {};
+    mutants.forEach(mutant => {
```

Comment:
I see what your doing here, but can't we the filtering already on line 121? That would save us the hassle of first generating the mutants and later grouping them.